src/transports/ws.js

```diff
@@ -0,0 +1,289 @@
+// Copyright 2024 Tether Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict'
+
+/**
+ * @typedef {Object} ElectrumWsConfig
+ * @property {string} url - The WebSocket URL (e.g., 'wss://electrum.example.com:50004').
+ */
+
+/** @typedef {import('./electrum-client.js').default} IElectrumClient */
+/** @typedef {import('./electrum-client.js').ElectrumBalance} ElectrumBalance */
+/** @typedef {import('./electrum-client.js').ElectrumUtxo} ElectrumUtxo */
+/** @typedef {import('./electrum-client.js').ElectrumHistoryItem} ElectrumHistoryItem */
+
+const isNode =
+  typeof process !== 'undefined' &&
+  !!(process.versions && process.versions.node)
+
+const WS_SPEC = 'ws'
+const WebSocket =
+  globalThis.WebSocket ??
+  (isNode ? (await import(/* @vite-ignore */ WS_SPEC)).default : undefined)
+if (!WebSocket) {
+  throw new Error('No WebSocket implementation available in this environment.')
+}
+
```

Comment:
```suggestion
const isNodeOrBare =
  typeof Bare !== 'undefined' ||
  (typeof process !== 'undefined' && !!process.versions?.node)


const WS_SPEC = 'ws'
const WebSocket =
  globalThis.WebSocket ??
  (isNodeOrBare ? (await import(/* @vite-ignore */ WS_SPEC)).default : undefined)
if (!WebSocket) {
  throw new Error('No WebSocket implementation available in this environment.')
}
```

Reply:
Addressed this in https://github.com/tetherto/wdk-wallet-btc/pull/55/commits/940ee322434d194007a30833bacc3507590da164